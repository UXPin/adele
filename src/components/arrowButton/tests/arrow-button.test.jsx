import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import ArrowButton from '../arrow-button';

const mockFunc = () => true;

it('renders arrowButton left correctly', () => {
  const wrapper = mount(<ArrowButton action={() => mockFunc()} direction="left" scrollerInactive="left" />);
  expect(wrapper).toMatchSnapshot();
});

it('renders arrowButton right correctly', () => {
  const wrapper = mount(<ArrowButton action={() => mockFunc()} direction="right" scrollerInactive="right" />);
  expect(wrapper).toMatchSnapshot();
});
