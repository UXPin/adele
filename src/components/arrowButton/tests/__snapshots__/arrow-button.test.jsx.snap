// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders arrowButton left correctly 1`] = `
.c1:focus {
  outline: 0;
}

.c1 > svg {
  width: 13px;
  height: 100%;
  fill: #777777;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  cursor: pointer;
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: none;
  outline: none;
  padding: 0;
}

.c0:focus > span > svg {
  fill: #006cff;
}

<ArrowButton
  action={[Function]}
  direction="left"
  scrollerInactive="left"
>
  <styled.button
    direction="left"
    onClick={[Function]}
    tabIndex={1}
    title="Move Table to the left"
  >
    <button
      className="c0"
      direction="left"
      onClick={[Function]}
      tabIndex={1}
      title="Move Table to the left"
    >
      <Icon
        active={false}
        color="#444"
        i="test-file-stub"
        in="blue"
        rotate={90}
        size="s"
      >
        <Styled(SVGInline)
          color="#444"
          data-active={false}
          in="blue"
          rotate={90}
          size="s"
          svg="test-file-stub"
        >
          <SVGInline
            className="c1"
            classSuffix="-svg"
            cleanup={Array []}
            cleanupExceptions={Array []}
            color="#444"
            component="span"
            data-active={false}
            in="blue"
            rotate={90}
            size="s"
            svg="test-file-stub"
          >
            <span
              className="SVGInline c1"
              color="#444"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
              data-active={false}
              in="blue"
              rotate={90}
              size="s"
            />
          </SVGInline>
        </Styled(SVGInline)>
      </Icon>
    </button>
  </styled.button>
</ArrowButton>
`;

exports[`renders arrowButton right correctly 1`] = `
.c1:focus {
  outline: 0;
}

.c1 > svg {
  width: 13px;
  height: 100%;
  fill: #777777;
  -webkit-transform: rotate(270deg);
  -ms-transform: rotate(270deg);
  transform: rotate(270deg);
  cursor: pointer;
}

.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  border: none;
  outline: none;
  padding: 0;
}

.c0:focus > span > svg {
  fill: #006cff;
}

<ArrowButton
  action={[Function]}
  direction="right"
  scrollerInactive="right"
>
  <styled.button
    direction="right"
    onClick={[Function]}
    tabIndex={1}
    title="Move Table to the right"
  >
    <button
      className="c0"
      direction="right"
      onClick={[Function]}
      tabIndex={1}
      title="Move Table to the right"
    >
      <Icon
        active={false}
        color="#444"
        i="test-file-stub"
        in="blue"
        rotate={270}
        size="s"
      >
        <Styled(SVGInline)
          color="#444"
          data-active={false}
          in="blue"
          rotate={270}
          size="s"
          svg="test-file-stub"
        >
          <SVGInline
            className="c1"
            classSuffix="-svg"
            cleanup={Array []}
            cleanupExceptions={Array []}
            color="#444"
            component="span"
            data-active={false}
            in="blue"
            rotate={270}
            size="s"
            svg="test-file-stub"
          >
            <span
              className="SVGInline c1"
              color="#444"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
              data-active={false}
              in="blue"
              rotate={270}
              size="s"
            />
          </SVGInline>
        </Styled(SVGInline)>
      </Icon>
    </button>
  </styled.button>
</ArrowButton>
`;
